vsts/gradle_build.cmd

```diff
@@ -9,4 +9,15 @@ if "%TARGET_BRANCH%" == "$(System.PullRequest.TargetBranch)" (SET isPullRequestB
 cd %build-root%\iot-e2e-tests\android
 call gradle wrapper --gradle-version 6.8.3
 call gradlew :clean :app:clean :app:assembleDebug
+
+if %ERRORLEVEL% NEQ 0 (
+  echo "Gradle build failed with error code %ERRORLEVEL%"
+  exit 1
```

Comment:
Previously, the gradle build could fail, but the ADO task would be green because we weren't manually returning a failed status code here